pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1023,7 +1024,11 @@ public void start() throws PulsarServerException {
             this.metricsGenerator = new MetricsGenerator(this);
 
             // the broker is ready to accept incoming requests by Pulsar binary protocol and http/https
-            readyForIncomingRequestsFuture.complete(null);
+            synchronized (pendingTasksBeforeReadyForIncomingRequests) {
+                pendingTasksBeforeReadyForIncomingRequests.forEach(Runnable::run);
+                pendingTasksBeforeReadyForIncomingRequests.clear();
+                readyForIncomingRequestsFuture.complete(null);
```

Comment:
Since the CompletableFuture features aren't needed anymore, this could be simply a boolean field.

Reply:
No, we need to wait the future being completed in `waitUntilReadyForIncomingRequests`.

```java
            public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
                    throws IOException, ServletException {
                try {
                    // Wait until the PulsarService is ready to serve incoming requests
                    pulsarService.waitUntilReadyForIncomingRequests();
```